### DISCUSSION TITLE
Directive Types

### DISCUSSION THREAD

**Discussion by TTWNO** [jcornaz/beancount-parser#5] [Q&A]
Currently the only associated Directive type is a transaction.

At minimum, there should be three additional directive types:

1.Account open & close.
2. Options. (Metadata, usually titles, precision display options, etc.)

I'm starting work on what will become `beancount-rs` and I need to verify that accounts are open.

**Comment by jcornaz** [comment]
Yes. Eventually, we may support everything in the beancount syntax. But I add things as they are needed for a use-case, and I wait for the need to actually arise first.

Obviously, a rewrite of `beancount` in rust would need everything by definition :smile:.

Personally, my priority is to first support tags for transactions, then the [price](https://beancount.github.io/docs/beancount_language_syntax.html#prices) directive. After that, my priority will (probably) be to improve the error handling to make clearer what is the problem in case of incorrect syntax.

I have personally no use-case for open, close and options. That doesn't mean I won't support them. I will probably implement them sooner or later (they should be very easy to implement anyway) it is just not my priority yet. If you want them to be supported quickly you can contribute them ;-)

I convert this into a discussion, as I feel it is not actionable (it kind of say "support the rest of the beancount syntax"  :smile:). Feel free to create more specific issues (like "support the `open` directive").

**Comment by TTWNO** [comment]
Fantastic! So glad to see that there is a solid outlook for this library!
I'd love to contribute my part. I'm not very well versed in `nom`, but I would certainly like to contribute.

**Reply by jcornaz** [reply]
Sure. I found nom is relatively simple and easy to learn. And hopefully, the other parsers and tests I wrote may also be helpful references. You can of course ask me questions about nom too ;-)

Otherwise don't worry, I'll probably still add the `open` and `close` directives soon-ish. As you asked for them and explained your use-case, I consider that the need has arisen, even if not for me. :smiley:

**Reply by jcornaz** [reply]
If you want to know more about my priorities, I created issues for the next features to add, and I also created this project: https://github.com/users/jcornaz/projects/6 to sort them by priority.

It is not intended to contain everything. Again, my approach is to only add items when I find that I need them. I am not eagerly implementing everything without purpose :-p

Thanks for your interest in this project :smiley:
